### ISSUE TITLE
optional varibles to specify credentials for backend bucket

### ISSUE THREAD

**Issue by rochana-atapattu** [suzuki-shunsuke/tfaction#1634]
### Feature Overview

just like we have variables in terraform plan/apply config for authentication, we can specify optional variables to configure backend authentication

### Why is the feature needed?

this optional variables allow user to manage state in different location (gcp project) and access to them separately. 

### Example Code

tfaction-root.yaml
exsisting
```yaml
target_groups:
 -  working_directory: gcp/
    target: gcp/
    gcs_bucket_name_tfmigrate_history: '<GCS Bucket Name for tfmigrate history files>'
    template_dir: templates/gcp
    terraform_plan_config:
      gcp_service_account: terraform-plan@example.iam.gserviceaccount.com
      gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    tfmigrate_plan_config:
      gcp_service_account: terraform-plan@example.iam.gserviceaccount.com
      gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    terraform_apply_config:
      gcp_service_account: terraform-apply@example.iam.gserviceaccount.com
      gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
    tfmigrate_apply_config:
      gcp_service_account: terraform-apply@example.iam.gserviceaccount.com
      gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
```
proposed
```yaml
target_groups:
 -  working_directory: gcp/
    target: gcp/
    gcs_bucket_name_tfmigrate_history: '<GCS Bucket Name for tfmigrate history files>'
    template_dir: templates/gcp
    terraform_plan_config:
      gcp_service_account: terraform-plan@example.iam.gserviceaccount.com
      gcp_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
      gcp_backend_service_account: terraform-plan@example.iam.gserviceaccount.com
      gcp_backend_workload_identity_provider: projects/000000000000/locations/global/workloadIdentityPools/github/providers/tfaction
```

**Comment by suzuki-shunsuke** [comment]
Thank you for your proposal.

GCS backend supports configuration options such as [impersonate_service_account](https://developer.hashicorp.com/terraform/language/settings/backends/gcs#impersonate_service_account).
Can you use these options?

https://developer.hashicorp.com/terraform/language/settings/backends/gcs

If we support variables you propose, how should tfaction use them?
Could you show us commands or GitHub Actions you expect?

**Comment by rochana-atapattu** [comment]
#1641/#1643 implements this feature. 
we can use the variables to generate a separate credentials file for backend config and provide it to terraform with an env var.

**Comment by suzuki-shunsuke** [comment]
Could you explain your motivation more detail?
Why do you want to use a different service account for Terraform Backend with service accounts for terraform plan / apply?

I think this is because you want to grant the least privileges to service accounts for terraform plan / apply for security.
For example, It would prevent malicious Terraform Providers from accessing Terraform State.
Is that correct?

Do you have any reference such as best practice related to this proposal?

**Comment by suzuki-shunsuke** [comment]
This issue was resolved by #1643 and v1.5.0 🎉
https://github.com/suzuki-shunsuke/tfaction/releases/tag/v1.5.0
